src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,174 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size == table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = calculateIndex(key);
+        if (table[index] == null) {
+            Node<K, V> newNode = new Node<>(key, value);
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if (isEqual(key, current.key)) {
+                    current.value = value;
+                    return;
+                }
+                current = current.next;
+            }
+            if (isEqual(key, current.key)) {
+                current.value = value;
+                return;
+            }
+            current.next = new Node<>(key, value);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = calculateIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (isEqual(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public V remove(Object key) {
+        Node<K, V> node = removeNode(key);
+        return node == null ? null : node.value;
+    }
+
+    @Override
+    public boolean containsKey(K key) {
+        int index = calculateIndex(key);
+        Node<K,V> currentNode = table[index];
+        while (currentNode != null) {
+            if (isEqual(key, currentNode.key)) {
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean containsValue(V value) {
+        if (table != null && size > 0) {
+            for (Node<K, V> currentNode : table) {
+                if (currentNode != null) {
+                    while (currentNode != null) {
+                        if (isEqual(value, currentNode.value)) {
+                            return true;
+                        }
+                        currentNode = currentNode.next;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private int hash(Object key) {
+        int hash;
+        return (key == null) ? 0 : (hash = key.hashCode()) ^ (hash >>> 16);
+    }
+
+    private int calculateIndex(Object key) {
+        return hash(key) & (table.length - 1);
```

Comment:
```suggestion
        return hash(key) & (table.length);
```

Reply:
As I uderstand if delete (-1) it won't work.

![22](https://user-images.githubusercontent.com/114337016/199568078-7773e0f7-b58d-436b-8534-c9ca35fc4866.png)